### PR TITLE
[Android][Feature][56522] feature/create_detail_screen 

### PR DESCRIPTION
**Pull request by hungvv-1301**
## Related Tickets
- [#Feature 56526: Detail Book](https://edu-redmine.sun-asterisk.vn/issues/56526)
- [#Feature 56527: View book detail and Save to favorite](https://edu-redmine.sun-asterisk.vn/issues/56527)
- [#Feature 56528: Add Book to Cart](https://edu-redmine.sun-asterisk.vn/issues/56528)
## WHAT
- 
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

**Review by nhanviet1 — APPROVED** [review]

File: app/src/main/java/com/atom/android/lebo/data/repository/book/BookRepository.kt [hunk 1]
```diff
@@ -24,4 +24,8 @@ class BookRepository(private val remote: IBookDataSource.Remote) : IBookReposito
     override fun getBookByID(id: Int): Single<BaseResponse<Book?>> {
         return remote.getBookByID(id)
     }
+
+    override fun getBookRelate(idBook: Int): Single<BaseResponse<List<Book>?>> {
+        return remote.getRelate(idBook)
+    }
```
Comment (hungld-1302) [inline app/src/main/java/com/atom/android/lebo/data/repository/book/BookRepository.kt]: override fun getBookRelate(idBook: Int): Single<BaseResponse<List<Book>?>> = 
        remote.getRelate(idBook)

File: app/src/main/java/com/atom/android/lebo/data/local/CartEntityLocal.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.atom.android.lebo.data.local
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Entity(tableName = DataBaseLocal.TABLE_CART)
+class CartEntityLocal(
+    @PrimaryKey
+    val idBook: Int,
+    val amount: Int = 1,
```
Comment (nguyennhatminh230801) [inline app/src/main/java/com/atom/android/lebo/data/local/CartEntityLocal.kt]: extract const nhé

**Comment by vietnh-1109** [comment]
Sao padding left lộn xộn thế em 
image

File: app/src/main/java/com/atom/android/lebo/ui/detail/adapter/ListAdapterBookVertical.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import android.view.ViewGroup
+import com.atom.android.lebo.base.BaseAdapter
+import com.atom.android.lebo.base.BaseViewHolder
+import com.atom.android.lebo.databinding.ItemBookVerticalBinding
+import com.atom.android.lebo.model.Book
+import com.atom.android.lebo.utils.extensions.loadImage
+
+class ListAdapterBookVertical(private val onClick: (Book) -> Unit) :
+    BaseAdapter<Book, BaseViewHolder<Book>>(Book.diffUtil) {
+
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/ui/detail/adapter/ListAdapterBookVertical.kt:14]: remove blank

File: app/src/main/java/com/atom/android/lebo/ui/favorite/FavoriteFragment.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    override val viewModel by viewModel<FavoriteViewModel>()
+
+    private val bookAdapter by lazy {
+        ListAdapterFavorite(onClick = { book ->
+            val action =
+                FavoriteFragmentDirections
+                    .actionNavigationFavoriteToNavigationDetail(book.idBook.toString())
+            findNavController().navigate(
+                action
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/ui/favorite/FavoriteFragment.kt]: tạo reference function ra em (::method) , ko viết thế này

**Pull request merged**